base/collection.go

```diff
@@ -468,6 +468,19 @@ func (b *GocbV2Bucket) QueryEpsCount() (int, error) {
 	return len(agent.N1qlEps()), nil
 }
 
+// GSIEps returns the GSI endpoints for the bucket for querying information about indexes.
+func (b *GocbV2Bucket) GSIEps() (url []string, err error) {
```

Comment:
Requires access to an internal port (9102), might want to call that out for future usage (currently test-only).